src/test/java/part1/exercise/StreamsExercise2.java

```diff
@@ -3,46 +3,138 @@
+        final Stream<PersonEmployerPair> personEmployerPairStream = getEmployees().stream()
+                .flatMap(StreamsExercise2::employeeToPersonEmployerPair);
+
+        final Map<String, List<Person>> employersStuffLists = personEmployerPairStream.collect(Collectors.groupingBy(
+                PersonEmployerPair::getEmployee,
+                Collectors.mapping(PersonEmployerPair::getPerson, Collectors.toList())
+        ));
+
+        final List<Person> expectedStuffLists_abc =
+                Arrays.asList(new Person("John", "Doe", 21),
+                            new Person("John", "Doe", 24),
+                            new Person("Bob", "Doe", 27),
+                            new Person("John", "Doe", 30));
+
+        Assert.assertEquals(expectedStuffLists_abc, employersStuffLists.get("abc"));
+    }
+
+    private static Stream<PersonEmployerPair> employeeToPersonEmployerPair(Employee employee) {
+        return employee.getJobHistory().stream()
+                .map(jobHistoryEntry -> new PersonEmployerPair(employee.getPerson(), jobHistoryEntry.getEmployer()));
     }
 
     @Test
     public void indexByFirstEmployer() {
-        Map<String, List<Person>> employeesIndex = null;// TODO
-        throw new UnsupportedOperationException();
+        final Stream<PersonEmployerPair> personFirstEmployerPairStream = getEmployees().stream()
+                .map(e -> new PersonEmployerPair(e.getPerson(), e.getJobHistory().get(0).getEmployer()));
+
+        final Map<String, List<Person>> employeesIndex = personFirstEmployerPairStream.collect(
+                Collectors.groupingBy(
+                        PersonEmployerPair::getEmployee,
+                        Collectors.mapping(PersonEmployerPair::getPerson, Collectors.toList())
+                ));
+
+        final List<Person> expectedListPersonWithFirstEmployer_yandex =
+                Arrays.asList(new Person("John", "Doe", 21),
+                        new Person("John", "Doe", 24),
+                        new Person("Bob", "Doe", 27),
+                        new Person("John", "Doe", 30));
+
+        Assert.assertEquals(expectedListPersonWithFirstEmployer_yandex, employeesIndex.get("yandex"));
+    }
+
+    private static class PersonEmployerDuration {
+        private final Person person;
+        private final String employer;
+        private final int duration;
+
+        public PersonEmployerDuration(Person person, String employer, int duration) {
+            this.person = person;
+            this.employer = employer;
+            this.duration = duration;
+        }
+
+        public Person getPerson() {
+            return person;
+        }
+
+        public String getEmployer() {
+            return employer;
+        }
+
+        public int getDuration() {
+            return duration;
+        }
     }
 
     @Test
     public void greatestExperiencePerEmployer() {
-        Map<String, Person> employeesIndex = null;// TODO

```

Comment:
Note that you should prefer person with 10 records with a single year `JobHistoryEntry` over person with 1 record with a single 5 years record.